ttnn/cpp/ttnn/operations/data_movement/permute/permute.cpp

```diff
@@ -193,23 +175,27 @@ ttnn::Tensor ExecutePermute::invoke(
     const ttnn::Tensor& input_tensor,
     tt::stl::Span<const int64_t> dims,
     const std::optional<MemoryConfig>& memory_config,
-    bool composite,
     const std::optional<float>& pad_value) {
-    if (composite) {
-        return detail::composite_invoke(input_tensor, dims, memory_config, pad_value);
-    }
-
-    const bool initial_input_tensor_on_device = detail::is_on_device(input_tensor);
-    const auto input_layout = input_tensor.get_layout();
     const auto input_rank = input_tensor.get_logical_shape().rank();
-
     TT_FATAL(
         input_rank == dims.size(),
         "The number of dimensions in the tensor input does not match the length of the desired ordering");
+    TT_FATAL(detail::is_on_device(input_tensor), "Tensor must already be on device");
+
+    SmallVector<uint32_t> normalized_dims(dims.size());
+    std::transform(dims.begin(), dims.end(), normalized_dims.begin(), [input_tensor](std::int64_t idx) {
+        return input_tensor.get_logical_shape().get_normalized_index(idx);
+    });
+    if (detail::is_permute_nop(input_tensor, normalized_dims)) {
```

Comment:
I think about a 0 volume shapes with an arbitary rank like `[1, 0, 32, 32]`. Will this be handled nicely? Can I ask for a test?

And I think about rank < 2, will this be handled correctly?

Reply:
for the 0 volume - there is no need for a device operation, but shape must be adjusted

Reply:
I return it as a nop when it's rank < 2 now (not just rank 1).

ttnn::empty doesn't have SimpleShape support right now which is a bit annoying. I have to think of and set my padded shape for the 0 logical volume case as well as the normal shape. For now, it seems like with device operation launch it still works, though could boost perf by a bit if we avoided it all togther.